rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -778,15 +770,13 @@ async fn run_controller(
                         tracing::error!("Crashing on purpose");
                         unsafe { sadness_generator::raise_segfault() }
                     },
-                    Req::Fail(Failure::Error) => bail!("Test error"),
+                    Req::Fail(Failure::Error) => Err(anyhow!("Test error"))?,
```

Comment:
I tried changing it back to `bail!` but it won't compile since `gui::Error` wraps `anyhow::Error` and the `bail!` doesn't seem to automatically do `into` to convert `anyhow::Error` to `gui::Error::Other(anyhow::Error)`